dfkernel/ipkernel.py

```diff
@@ -101,6 +128,15 @@ async def execute_request(self, stream, ident, parent):
 
         input_tags = dfkernel_data.get("input_tags", {})
         # print("SETTING INPUT TAGS:", input_tags, file=sys.__stdout__)
+
+        output_tags= {}
```

Comment:
I would probably use defaultdict here instead of setdefault, it's a little easier for the sake of readability

Reply:
I agree that defaultdict is better here, but be careful with defaultdict. When I am constructing a dict, I use it, but then I convert it to a regular dict when accessing it.

Reply:
Update code with defaultdict and before parsing converted to regular a dict